company-flow.el

```diff
@@ -0,0 +1,91 @@
+;;; company-flow.el --- Flow backend for company-mode  -*- lexical-binding: t -*-
+(require 'cl-lib)
+
+(defun company-flow-handle-signal (process _event)
+  (when (memq (process-status process) '(signal exit))
+    (let ((callback (process-get process 'company-flow-callback))
+          (prefix (process-get process 'company-flow-prefix)))
+      ;; (message "%S" (company-flow-parse-output (company-flow-get-output process)))
+      (funcall callback (->> process
+                             company-flow-get-output
+                             company-flow-parse-output
+                             (all-completions prefix))))))
+
+(defun company-flow-make-candidate (line)
+  (let* ((words (split-string line " "))
+         (text (car words))
+         (meta (mapconcat 'identity (cdr words) " ")))
+    (propertize text 'meta meta)))
+
+(defun company-flow-parse-output (output)
+  (mapcar 'company-flow-make-candidate
```

Comment:
> There isn't an output buffer at all in this, just a string that gets passed around.

There is. The process is associated with a buffer. You're just using `process-filter` (which gets passed the output string(s)), so you don't see that.

The alternative is parsing the output buffer contents from the sentinel.

> Are you thinking that putting it into a buffer then re-search-forwarding through it would be faster or more efficient?

Same thing?

> I can see it being more memory efficient possibly because the split list doesn't exist, but I don't understand enough about emacs perf or memory to know if it'd be all that much better.

Yes, naturally you'll have to measure first, before tweaking. Evaluating `(benchmark 1 '(let ((company-backend 'company-flow)) (company-call-backend 'candidates "")))` should do that, but you might need to find a big enough project, so that the time is noticeable (or not).